tests/datasets/causal_data/test_causal_utility_functions.py

```
@@ -0,0 +1,166 @@
+
+            # Clear all datasets
+            clear_causal_data()
+            assert not os.path.exists(os.path.join(temp_dir, 'diamond'))
+            assert not os.path.exists(os.path.join(temp_dir, 'fork'))
+
+
+def test_list_available_datasets():
+    """Test the list_available_datasets function."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Patch the default directory to use temporary directory
+        with patch('os.path.expanduser', return_value=temp_dir):
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)